### PR TITLE
Add new field `auto_rotate_period` on transit key management

### LINKED ISSUE #902: Implement the new field `auto_rotate_period` on transit function
Ref documentation: https://developer.hashicorp.com/vault/api-docs/secret/transit#auto_rotate_period

Can we have the implementation of the new `auto_rotate_period` field related to the documentation?

To be implemented on the `create_key` and `update _key`

### PR DESCRIPTION
**Pull request by sebglon**
Implement new `auto_rorate_period` on transit `create_key` and `update_key`

Resolves #902

### CODE REVIEW

**Comment by colin-pm** [comment]
Change overall LGTM, thanks for the contribution!

Can you run black and amend your commit?

@adammike @briantist This field appears to be added in v1.10.x. Thankfully it looks like using the parameter doesn't break the integration tests for <1.10.x, however should it be on us to mark when these fields were added?

**Comment by briantist** [comment]
> This field appears to be added in v1.10.x. Thankfully it looks like using the parameter doesn't break the integration tests for <1.10.x, however should it be on us to mark when these fields were added?

Yes, we don't have to be perfect about it but when possible we should mark when a specific documented field was added to Vault, especially if it does not pre-date the versions of Vault we officially support (but we should still seek to do it even if it does).

**Comment by sebglon** [comment]
@colin-pm Format is fixed
Do you need more change on my side?

**Review by briantist — DISMISSED** [review]

File: hvac/api/secrets_engines/transit.py [hunk 1]
```diff
@@ -59,6 +60,8 @@ def create_key(
         :type key_type: str | unicode
         :param mount_point: The "path" the method/backend was mounted on.
         :type mount_point: str | unicode
+        :param auto_rotate_period: The period at which this key should be rotated automatically.
```
Comment (briantist) [inline hvac/api/secrets_engines/transit.py:63]:
```suggestion
        :param auto_rotate_period: The period at which this key should be rotated automatically. Requires Vault 1.10.x.
```

File: hvac/api/secrets_engines/transit.py [hunk 2]
```diff
@@ -201,6 +206,8 @@ def update_key_configuration(
         :type allow_plaintext_backup: bool
         :param mount_point: The "path" the method/backend was mounted on.
         :type mount_point: str | unicode
+        :param auto_rotate_period: The period at which this key should be rotated automatically.
```
Comment (briantist) [inline hvac/api/secrets_engines/transit.py:209]:
```suggestion
        :param auto_rotate_period: The period at which this key should be rotated automatically. Requires Vault 1.10.x.
```

**Review by briantist — APPROVED** [review]

**Comment by briantist** [comment]
@sebglon thanks very much for this contribution!

**Pull request merged**